src/util/message.h

```diff
@@ -154,10 +154,17 @@ class messaget
 
   messaget(const messaget &other):
     message_handler(other.message_handler),
-    mstream(other.mstream)
+    mstream(other.mstream, *this)
   {
   }
 
+  messaget &operator=(const messaget &other)
+  {
+    message_handler=other.message_handler;
+    mstream.assign_from(other.mstream);
+    return *this;
+  }
```

Comment:
Why are you deleting copy-constructor and adding copy-assignment? Either none or both should be available.

Reply:
I'm not. `messaget`'s copy-constructor is fixed and gains a matching `operator=`; its internal class `mstreamt` is made neither copyable nor assignable (it should always move around with its enclosing messaget)